lizmap/modules/lizmap/lib/Project/QgisProject.php

```diff
@@ -1362,4 +1384,234 @@ protected function readLayers($xml)
+            $fieldEditType = (string) $editWidget->attributes()->type;
+            $options = $editWidget->config->Option;
+            $edittypes[$fieldName] = array();
+            // Option + Attributes
+            if (count((array) $options) > 2) {
+                \jLog::log('More than one Option found in the Qgis File for field '.$fieldName.', only the first will be read.', 'warning');
+            }
+            $fieldEditOptions = array();
+            foreach ($options->Option as $option) {
+                if ((string) $option->attributes()->type === 'List') {
+                    foreach ($option->Option as $l) {
+                        if ((string) $l->attributes()->type === 'Map') {
+                            $values = $this->getValuesFromOptions($l);
+                        } else {
+                            $values = $this->getValuesFromOptions($l, false);
+                        }
+                    }
+                    $fieldEditOptions[(string) $option->attributes()->name] = $values;
+                // Option with list of values as Map
+                } elseif ((string) $option->attributes()->type === 'Map') {
+                    $fieldEditOptions[(string) $option->attributes()->name] = $this->getValuesFromOptions($option);
+
+                // Option with string list of values
+                } elseif ((string) $option->attributes()->type === 'StringList') {
+                    $fieldEditOptions[(string) $option->attributes()->name] = ${$this}->getValuesFromOptions($option, false);
+                // Simple option
+                } else {
+                    $fieldEditOptions[(string) $option->attributes()->name] = (string) $option->attributes()->value;
+                }
+            }
+
+            $edittype = array(
+                'type' => $fieldEditType,
+                'options' => (object) $fieldEditOptions,
+            );
+
+            // editable
+            $editableFieldXml = $layerXml->xpath("editable/field[@name='${fieldName}']");
+            if ($editableFieldXml && count($editableFieldXml)) {
+                $edittype['editable'] = (int) $editableFieldXml[0]->attributes()->editable;
+            } else {
+                $edittype['editable'] = 1;
+            }
+
+            $edittype = (object) $edittype;
+            $edittypes[$fieldName]['edittype'] = $edittype;
+            $edittypes[$fieldName]['widgetv2configAttr'] = $edittype->options;
+            $edittypes[$fieldName]['fieldEditType'] = $edittype->type;
+        }
+
+        return $edittypes;
+    }
+
+    protected function getEditType($layerXml)
+    {
+        $edittypes = $layerXml->edittypes;
+        $editTab = array();
+
+        foreach ($edittypes as $edittype) {
+            $attributes = $edittype->attributes();
+            $editTab[$attributes->name] = array();
+            $editTab[$attributes->name]['eddittype'] = $edittype;
+            if (property_exists($edittype->attributes(), 'widgetv2type')) {
```

Comment:
an original comment is missing :  `// New QGIS 2.4 edittypes : use widgetv2type property`